### ISSUE TITLE
select columns from an insert

### ISSUE THREAD

**Issue by andykais** [supabase/postgrest-js#143]
# Feature request

## Is your feature request related to a problem? Please describe.
I need a way to perform an upsert, and know if I performed an update or an insert afterwards. Postgrest has an open issue to support this via 200 and 201 status codes https://github.com/PostgREST/postgrest/issues/1070. For the time being though, I can infer this on the client side using the `xmax` column. E.g.
```
POST http://<supabase-db-url>.supabase.co/rest/v1/items?select=id,xmax
```
If `xmax === "0"` then the row was inserted, otherwise an update occurred. Unfortunately, there is no way to write this kind of query with the supabase client.

## Describe the solution you'd like
```js
const result = await supabase
  .from('items')
  .insert([row], { onConflict: 'unique_field', upsert: true })
  .select('id,xmax')
if (result.data[0].xmax === "0") {
  console.log('insert')
} else {
  console.log('update')
}
```

## Describe alternatives you've considered
write an rpc function to wrap this logic

## Additional context
an issue on the PostgREST repo discussing possible solutions here: https://github.com/PostgREST/postgrest/issues/1683

selects on insert is valuable for multiple reasons outside of my outlined use case. Any performant client will be interested in returning the minimal rows necessary after an update/insert, rather than all of them all the time. I feel like I saw an issue or discussion around the default behavior around the `RETURNING` clause but for the life of me I cannot find it.

**Comment by kiwicopple** [comment]
We now have `returning=minimal|representation` - https://github.com/supabase/postgrest-js/issues/127 - but it isn't going to solve `xmax`.

To support this, it [looks like](https://github.com/PostgREST/postgrest/issues/1683#issuecomment-740661858) we just need to append a `select` onto the POST request. 

It might be tricky to support this syntax:

```js
await supabase
  .from('items')
  .insert([row], { onConflict: 'unique_field', upsert: true })
  .select('id,xmax')
```

because the promise fires after the `insert()`, but I think we could do something like this:

```js
await supabase
  .from('items')
  .insert([row], { 
    onConflict: 'unique_field', 
    upsert: true,
    returning: 'id,xmax'
  })
```

This would extend the work we have done on https://github.com/supabase/postgrest-js/issues/127. If it's `minimal | representation` then we pass it as a header, but if it's anything else, then we append the `select=...` onto the URL. Would that work @steve-chavez, @soedirgo  ?

**Comment by soedirgo** [comment]
I think we can do
```js
await supabase
  .from('items')
  .insert([row], { onConflict: 'unique_field', upsert: true })
  .select('id,xmax')
```
By returning `PostgrestQueryBuilder` after `INSERT`/`UPDATE`/`DELETE` instead of `PostgrestFilterBuilder`. Wdyt?
https://github.com/supabase/postgrest-js/blob/11d936612bd225ae493eb4bfd06a579cfa486fb2/src/lib/PostgrestQueryBuilder.ts#L52-L74

**Comment by soedirgo** [comment]
I thought I could just do `select()` after `insert()`, but that would overwrite `method` to `GET`, so that's a no go.

I'm not too comfortable with overloading `returning` with the columns, I prefer something like `returnedColumns: ...`. But if we do use `returning`, maybe it's better for it to be an array instead, so we can disambiguate whether `'minimal'` means don't do `RETURNING` or if it's a column named `minimal`.

**Comment by steve-chavez** [comment]
> Edit: Or we could check this.method in select() so we don't overwrite it if it's already set. This is the cleanest option IMO

@soedirgo I agree. It'd be better to support `select` because PostgREST can also embed after [insertions/updates/deletes](https://postgrest.org/en/v7.0.0/api.html#embedding-after-insertions-updates-deletions). And it can also [embed on rpc](https://postgrest.org/en/v7.0.0/api.html#embedding-on-stored-procedures).

So supporting `select` as a modifier would make all of that functionality more consistent:

```js
await supabase
  .from('items')
  .insert("{...}")
  .select('id,other,parent(id,name)')

await supabase
  .from('items')
  .update("{...}")
  .select('id,other,parent(id,name)')

await supabase
  .from('items')
  .delete("{...}")
  .select('id,other,parent(id,name)')

await supabase
  .rpc("myrpc", "{..}")
  .select('id,other,parent(id,name)')

**Comment by steve-chavez** [comment]
> minimal | representation 

One thing about `minimal` is that doing a `select` after a mutation won't give an error(according to [RFC](https://tools.ietf.org/html/rfc7240), `Prefer` headers can't err). So we could err on `postgrest-js` if `minimal` is used with a `select`.

**Comment by kiwicopple** [comment]
OK cool, Steve's [examples](https://github.com/supabase/postgrest-js/issues/143#issuecomment-750344219) actually are a very nice DX, but more importantly - since PostgREST supports embedding on things like `rpc` (TIL!), then we should definitely support it in `postgrest-js` too 👍

**Comment by SteveChurch** [comment]
@soedirgo sorry to comment on an old issue, but does this also exist on `upsert` - From looking at the code base I don't think it does?

**Comment by soedirgo** [comment]
No probs! Yes, it should work on `upsert`: https://supabase.com/docs/reference/javascript/upsert.
